common/metrics/metrics.go

```diff
@@ -102,3 +173,182 @@ func getMigrationDataSourceDetails(driver string, migrationData *migration.Migra
 		return migration.MigrationData_SOURCE_CONNECTION_MECHANISM_UNSPECIFIED.Enum(), migration.MigrationData_SOURCE_UNSPECIFIED.Enum()
 	}
 }
+
+// CreateDataflowShardMonitoringDashboard returns a monitoring dashboard for a single shard
+func CreateDataflowShardMonitoringDashboard(ctx context.Context, resourceIds MonitoringMetricsResources) (*dashboardpb.Dashboard, error) {
```

Comment:
This is a hot topic, but I personally think that this method would be more readable using `MonitoringMetricsResources` as a function receiver instead of a parameter